diff --git a/libs/date_time/test/Jamfile.v2 b/libs/date_time/test/Jamfile.v2
index 74da9b9..b3e25ce 100644
--- a/libs/date_time/test/Jamfile.v2
+++ b/libs/date_time/test/Jamfile.v2
@@ -3,7 +3,6 @@ import path ;
 import regex ;
 import testing ;
 
-local DATE_TIME_DYNAMIC_PROPERTIES = <define>BOOST_ALL_DYN_LINK <runtime-link>shared <define>BOOST_DATE_TIME_POSIX_TIME_STD_CONFIG <define>BOOST_ALL_NO_LIB ;
 local DATE_TIME_PROPERTIES = <define>BOOST_DATE_TIME_POSIX_TIME_STD_CONFIG
   # FIXME 
   #std::locale-support 
diff --git a/libs/filesystem/test/Jamfile.v2 b/libs/filesystem/test/Jamfile.v2
index 8dccb9f..e617e84 100644
--- a/libs/filesystem/test/Jamfile.v2
+++ b/libs/filesystem/test/Jamfile.v2
@@ -70,7 +70,6 @@ if [ os.environ UBSAN ]
         <library>/boost/filesystem//boost_filesystem/<visibility>global ;
 }
 
-run config_info.cpp : : : <link>shared <test-info>always_show_run_output ;
 run config_info.cpp : : : <link>static <test-info>always_show_run_output : config_info_static ;
 run convenience_test.cpp : : : <define>BOOST_FILESYSTEM_VERSION=4 ;
 compile macro_default_test.cpp ;
@@ -79,16 +78,11 @@ run deprecated_test.cpp : : : <define>BOOST_FILESYSTEM_VERSION=3 ;
 run fstream_test.cpp : : : $(VIS) <define>BOOST_FILESYSTEM_VERSION=4 ;
 run large_file_support_test.cpp : : : <define>BOOST_FILESYSTEM_VERSION=4 ;
 run locale_info.cpp : : : <define>BOOST_FILESYSTEM_VERSION=4 <test-info>always_show_run_output ;
-run operations_test.cpp : : : <link>shared <define>BOOST_FILESYSTEM_VERSION=4 <test-info>always_show_run_output ;
 run operations_test.cpp : : : <link>static <define>BOOST_FILESYSTEM_VERSION=4 : operations_test_static ;
-run operations_unit_test.cpp : $(HERE) : : <link>shared <define>BOOST_FILESYSTEM_VERSION=4 <test-info>always_show_run_output ;
+run operations_unit_test.cpp : $(HERE) : : <link>static <define>BOOST_FILESYSTEM_VERSION=4 <test-info>always_show_run_output ;
 run copy_test.cpp : : : <define>BOOST_FILESYSTEM_VERSION=4 ;
-run path_test.cpp : : : <link>shared <define>BOOST_FILESYSTEM_VERSION=4 ;
 run path_test.cpp : : : <link>static <define>BOOST_FILESYSTEM_VERSION=4 : path_test_static ;
-run path_test.cpp : : : <link>shared <define>BOOST_FILESYSTEM_VERSION=3 : path_test_v3 ;
-run path_unit_test.cpp : : : <link>shared $(VIS) <define>BOOST_FILESYSTEM_VERSION=4 ;
 run path_unit_test.cpp : : : <link>static $(VIS) <define>BOOST_FILESYSTEM_VERSION=4 : path_unit_test_static ;
-run path_unit_test.cpp : : : <link>shared $(VIS) <define>BOOST_FILESYSTEM_VERSION=3 : path_unit_test_v3 ;
 run relative_test.cpp : : : <define>BOOST_FILESYSTEM_VERSION=4 ;
 run ../example/simple_ls.cpp : : : <define>BOOST_FILESYSTEM_VERSION=4 ;
 run ../example/file_status.cpp : : : <define>BOOST_FILESYSTEM_VERSION=4 ;
diff --git a/libs/program_options/test/Jamfile.v2 b/libs/program_options/test/Jamfile.v2
index 0f1a6e7..617c8c9 100644
--- a/libs/program_options/test/Jamfile.v2
+++ b/libs/program_options/test/Jamfile.v2
@@ -4,7 +4,7 @@ project
     : requirements 
     <library>../build//boost_program_options
     <link>static
-    <variant>debug
+#   <variant>debug
     
 #    <define>_GLIBCXX_CONCEPT_CHECKS
 #    <define>_GLIBCXX_DEBUG
@@ -14,9 +14,6 @@ rule po-test ( source : input-file ? )
 {
     return
         [ run $(source) : : $(input-file) ]
-        [ run $(source) : : $(input-file) 
-          : <link>shared <define>BOOST_PROGRAM_OPTIONS_DYN_LINK=1
-          : $(source:B)_dll ] 
     ;   
 }    
 
diff --git a/libs/regex/test/Jamfile.v2 b/libs/regex/test/Jamfile.v2
index 87bc160..2885ba8 100644
--- a/libs/regex/test/Jamfile.v2
+++ b/libs/regex/test/Jamfile.v2
@@ -138,12 +138,6 @@ test-suite regex
          : # input files
          : <test-info>always_show_run_output
       ]
-      [ run config_info/regex_config_info.cpp ../build//boost_regex 
-        : # command line
-        : # input files
-        : <test-info>always_show_run_output
-        : regex_dll_config_info
-      ]    
 
       [ run collate_info/collate_info.cpp ../build//boost_regex
       : : : <test-info>always_show_run_output : test_collate_info ]
@@ -200,7 +194,7 @@ test-suite regex
     <define>BOOST_NO_EXCEPTIONS=1
     <exception-handling>off
     <link>static
-    <runtime-link>shared
+#   <runtime-link>shared
   : regex_regress_noeh ]
 
 ;
diff --git a/libs/system/test/Jamfile.v2 b/libs/system/test/Jamfile.v2
index 0a6bc3e..ff60879 100644
--- a/libs/system/test/Jamfile.v2
+++ b/libs/system/test/Jamfile.v2
@@ -28,7 +28,6 @@ rule system-run ( sources + )
 
     result += [ run $(sources) ] ;
     result += [ run $(sources) : : : <library>/boost/system//boost_system <link>static : $(sources[1]:B)_static ] ;
-    result += [ run $(sources) : : : <library>/boost/system//boost_system <link>shared : $(sources[1]:B)_shared ] ;
     result += [ run $(sources) : : : <define>BOOST_NO_ANSI_APIS : $(sources[1]:B)_no_ansi ] ;
     result += [ run $(sources) : : : <define>BOOST_SYSTEM_USE_UTF8 : $(sources[1]:B)_utf8 ] ;
 
@@ -41,7 +40,7 @@ system-run system_error_test.cpp ;
 
 lib throw_test : throw_test.cpp : <link>shared:<define>THROW_DYN_LINK=1 ;
 
-run dynamic_link_test.cpp throw_test : : : <link>shared : throw_test_shared ;
+run dynamic_link_test.cpp throw_test : : : <link>static : throw_test_shared ;
 
 system-run initialization_test.cpp ;
 system-run header_only_test.cpp ;
@@ -56,7 +55,6 @@ lib single_instance_lib2 : single_instance_2.cpp : <link>shared:<define>SINGLE_I
 
 system-run single_instance_test.cpp single_instance_1.cpp single_instance_2.cpp ;
 run single_instance_test.cpp single_instance_lib1 single_instance_lib2 : : : <link>static : single_instance_lib_static ;
-run single_instance_test.cpp single_instance_lib1 single_instance_lib2 : : : <link>shared : single_instance_lib_shared ;
 
 system-run before_main_test.cpp ;
 run-fail throws_assign_fail.cpp ;
@@ -84,7 +82,6 @@ lib std_single_instance_lib2 : std_single_instance_2.cpp : <link>shared:<define>
 
 system-run std_single_instance_test.cpp std_single_instance_1.cpp std_single_instance_2.cpp ;
 run std_single_instance_test.cpp std_single_instance_lib1 std_single_instance_lib2 : : : <link>static : std_single_instance_lib_static ;
-run std_single_instance_test.cpp std_single_instance_lib1 std_single_instance_lib2 : : : <link>shared <define>STD_SINGLE_INSTANCE_SHARED : std_single_instance_lib_shared ;
 
 run is_error_code_enum_test.cpp ;
 run is_error_condition_enum_test.cpp ;
