diff --git a/CMake/FollyCompilerUnix.cmake b/CMake/FollyCompilerUnix.cmake
index f10e1c3..0706119 100644
--- a/CMake/FollyCompilerUnix.cmake
+++ b/CMake/FollyCompilerUnix.cmake
@@ -28,7 +28,7 @@ set(
 )
 mark_as_advanced(CXX_STD)
 
-set(CMAKE_CXX_FLAGS_COMMON "-g -Wall -Wextra")
+set(CMAKE_CXX_FLAGS_COMMON "-Wall -Wextra")
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${CMAKE_CXX_FLAGS_COMMON}")
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${CMAKE_CXX_FLAGS_COMMON} -O3")
 
@@ -43,7 +43,6 @@ function(apply_folly_compile_options_to_target THETARGET)
   )
   target_compile_options(${THETARGET}
     PRIVATE
-      -g
       -std=${CXX_STD}
       -finput-charset=UTF-8
       -fsigned-char
diff --git a/CMake/folly-deps.cmake b/CMake/folly-deps.cmake
index 989259a..02faef8 100644
--- a/CMake/folly-deps.cmake
+++ b/CMake/folly-deps.cmake
@@ -70,6 +70,9 @@ find_package(LibEvent MODULE REQUIRED)
 list(APPEND FOLLY_LINK_LIBRARIES ${LIBEVENT_LIB})
 list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LIBEVENT_INCLUDE_DIR})
 
+set(FOLLY_ORIGINAL_CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS}")
+string(REGEX REPLACE " -std=[^ ]*" "" CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS}")
+
 find_package(ZLIB MODULE)
 set(FOLLY_HAVE_LIBZ ${ZLIB_FOUND})
 if (ZLIB_FOUND)
@@ -104,6 +107,8 @@ if (LIBLZMA_FOUND)
   list(APPEND FOLLY_LINK_LIBRARIES ${LIBLZMA_LIBRARIES})
 endif()
 
+set(CMAKE_REQUIRED_FLAGS "${FOLLY_ORIGINAL_CMAKE_REQUIRED_FLAGS}")
+
 find_package(LZ4 MODULE)
 set(FOLLY_HAVE_LIBLZ4 ${LZ4_FOUND})
 if (LZ4_FOUND)
diff --git a/CMake/libfolly.pc.in b/CMake/libfolly.pc.in
index ffa043c..039a102 100644
--- a/CMake/libfolly.pc.in
+++ b/CMake/libfolly.pc.in
@@ -1,5 +1,5 @@
 prefix=@CMAKE_INSTALL_PREFIX@
-exec_prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
 libdir=${exec_prefix}/@CMAKE_INSTALL_LIBDIR@
 includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
 
diff --git a/folly/io/async/test/EventBaseTestLib.h b/folly/io/async/test/EventBaseTestLib.h
index 7749a73..849ee82 100644
--- a/folly/io/async/test/EventBaseTestLib.h
+++ b/folly/io/async/test/EventBaseTestLib.h
@@ -2318,7 +2318,7 @@ TYPED_TEST_P(EventBaseTest1, DrivableExecutorTest) {
   folly::EventBase& base = *eventBasePtr;
   bool finished = false;
 
-  Baton baton;
+  Baton<> baton;
 
   std::thread t([&] {
     baton.wait();
diff --git a/folly/test/MemoryTest.cpp b/folly/test/MemoryTest.cpp
index 7de2a39..35c546e 100644
--- a/folly/test/MemoryTest.cpp
+++ b/folly/test/MemoryTest.cpp
@@ -76,7 +76,7 @@ TEST(make_unique, compatible_with_std_make_unique) {
 }
 
 TEST(to_shared_ptr_aliasing, example) {
-  auto sp = folly::copy_to_shared_ptr(std::tuple{3, 4});
+  auto sp = folly::copy_to_shared_ptr(std::tuple<int, int>{3, 4});
   auto a = folly::to_shared_ptr_aliasing(sp, &std::get<1>(*sp));
   EXPECT_EQ(4, *a);
 }
